oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServerOAuth2AuthorizedClientExchangeFilterFunction.java

```diff
@@ -303,11 +296,14 @@ private boolean shouldRefreshToken(OAuth2AuthorizedClient authorizedClient) {
 		if (this.authorizedClientRepository == null) {
 			return false;
 		}
+		if (authorizedClient.getClientRegistration().getAuthorizationGrantType().equals(AuthorizationGrantType.CLIENT_CREDENTIALS)) {
```

Comment:
Same comments as above - a `client_credentials` client would never have a refresh token and this method is logic specific to refresh token.